swan/scripts/before-notebook.d/02_environment.sh

```diff
@@ -0,0 +1,110 @@
+#!/bin/bash
+
+# Author: Danilo Piparo, Enric Tejedor, Pedro Maximino, Diogo Castro 2023
+# Copyright CERN
+# Here the environment for the notebook server is prepared. Many of the commands are 
+# launched as regular user as it's this entity which is able to access 
+# eos and not the super user.
+
+# Set environment for the Jupyter process
+LOCAL_HOME=/home/$NB_USER
+export IPYTHONDIR=$LOCAL_HOME/.ipython
+export JPY_DIR=$LOCAL_HOME/.jupyter
+
+# JUPYTER_CONFIG_DIR is set to point to the local user home (/home/$USER)
+# instead of the EOS path
+export JUPYTER_CONFIG_DIR=$JPY_DIR
+export JUPYTER_PATH=$LOCAL_HOME/.local/share/jupyter
+
+# Set other environment variables
+export KERNEL_DIR=$JUPYTER_PATH/kernels
+export SWAN_ENV_FILE=$LOCAL_HOME/.bash_profile
+export PROFILEPATH=$IPYTHONDIR/profile_default
+
+# Create missing directories
+mkdir -p $IPYTHONDIR $PROFILEPATH
+
+# Setup the LCG View on CVMFS
+_log "Setting up environment from CVMFS"
+export LCG_VIEW=$ROOT_LCG_VIEW_PATH/$ROOT_LCG_VIEW_NAME/$ROOT_LCG_VIEW_PLATFORM
+
+# symlink $LCG_VIEW/share/jupyter/nbextensions for the notebook extensions
+ln -s $LCG_VIEW/share/jupyter/nbextensions $JUPYTER_PATH
+#Creating a ROOT_DATA_DIR variable
+export ROOT_DATA_DIR=$(readlink $LCG_VIEW/bin/root | sed -e 's/\/bin\/root//g')
+
+cp -L -r $LCG_VIEW/etc/jupyter/* $JUPYTER_CONFIG_DIR
+
+# Configure kernels and terminal
+# The environment of the kernels and the terminal will combine the view and the user script (if any)
+_log "Configuring kernels and terminal"
+
+# ROOT
+ROOT_KERNEL_PATH=$LCG_VIEW/etc/notebook/kernels/root
+if [ -d $ROOT_KERNEL_PATH ];
+then
+  cp -rL $ROOT_KERNEL_PATH $KERNEL_DIR
+fi
+
+# R
+R_KERNEL_PATH=$LCG_VIEW/share/jupyter/kernels/ir
+if [ -d $R_KERNEL_PATH ];
+then
+  cp -rL $R_KERNEL_PATH $KERNEL_DIR
+  sed -i "s/IRkernel::main()/options(bitmapType='cairo');IRkernel::main()/g" $KERNEL_DIR/ir/kernel.json # Force cairo for graphics
+fi
+
+# Octave
+OCTAVE_KERNEL_PATH=$LCG_VIEW/share/jupyter/kernels/octave
+if [[ -d $OCTAVE_KERNEL_PATH ]];
+then
+   cp -rL $OCTAVE_KERNEL_PATH $KERNEL_DIR
+   export OCTAVE_KERNEL_JSON=$KERNEL_DIR/octave/kernel.json
+fi
+
+# Julia
+JULIA_KERNEL_PATH=$LCG_VIEW/share/jupyter/kernels/julia-*
+if [ -d $JULIA_KERNEL_PATH ];
+then
+  cp -rL $JULIA_KERNEL_PATH $KERNEL_DIR
+fi
+
+# Grant privileges to all files inside the created directories and subdirectoies
+# to the user
+chown -R $NB_USER:$NB_GID $LOCAL_HOME
+
+_log "Running user configuration script for user $NB_USER."
+sudo -E -u $NB_USER bash /srv/singleuser/scripts/userconfig.sh
+
+if [ $? -ne 0 ]
+then
+  _log "Error configuring user environment"
+  exit 1
+fi
+
+START_TIME_CONFIGURE_KERNEL_ENV=$( date +%s.%N )
+
+# Make sure we have a sane terminal
+printf "export TERM=xterm\n" >> $SWAN_ENV_FILE
+
+# If there, source users' .bashrc after the SWAN environment
+BASHRC_LOCATION=$LOCAL_HOME/.bashrc
```

Comment:
Do we need to source the `.bashrc` in the `LOCAL_HOME` that we just created? Is this meant to be the EOS home instead?

Reply:
Ah I see, yeah this is my mistake. It is supposed to be the EOS home and not the local home of the user.

Reply:
What I see now in production is that inside the SWAN_ENV_FILE, what we do is source the .bashrc inside the EOS HOME, not the LOCAL HOME. I would check, with this image, what is the .bashrc that is being sourced inside SWAN_ENV_FILE (should not be /home/username).

Reply:
Indeed this was a problem, @PMax5 fixed it, resolving.